src/frontend/Axis.hpp

```diff
@@ -219,6 +223,22 @@ class Axis : public Drawable {
   /// gets the number of ticks on this axis
   Vector<int, 2> get_ticks() const { return {m_nx_ticks, m_ny_ticks}; }
 
+  /// get the number of maximum yticks length (in pixels)
+  int max_ytick_pixels() const {
+    /// Convertion from length * font_size to pixels
+    /// The scale factor, i.e., 1.25, is for Roboto font (https://bugbox.clickteam.com/threads/82514-Useful-Roboto-Font-Size-Pixel-Heights!)
+    return int(this->max_ytick_char_len() * m_style.font_size() / 1.25);
+  }
+
+  // get the number of maximum char length of yticks
+  int max_ytick_char_len() const { return m_max_ytick_char_len; }
```

Comment:
`m_max_ytick_char_len` depends on if the drawing code is run yet or not. please remove this method and keep this value purely private to the class